pilot/pkg/model/service.go

```diff
@@ -388,6 +392,10 @@ func (si *ServiceInstance) GetLocality() string {
 	if si.Endpoint.Locality != "" {
 		return si.Endpoint.Locality
 	}
+	if strings.Contains(si.Labels[LocalityLabel], k8sSeparator) {
```

Comment:
I don't think we need the check, we can just replace them. Checking is probably just as slow as changing it when there is no match. Not a big deal though

Reply:
strings.Contains is absolutely more efficient than Replace. Can look at the implement

Reply:
First, I don't really care if we keep it or change it, I was just curious so I ran a quick benchmark. 

I don't think it is that simple. Of course contains is faster than replace, but replace is pretty fast IF we don't need to replace anything.


Code with contains (your code):
if we do need to replace, 140ns
if we don't, 10ns

Code without contains:
if we do need to replace, 120ns
if we don't, 13ns

So its actually probably faster to not have contains, because we will most commonly be replacing, so we just have an extra contains call. Replace is actually very fast if it doesn't do any replacing.

The only reason I mentioned it was because we are adding (a very small amount) of complexity to avoid doing a "costly" operation (replace), but any time that operation would actually be costly is when we would do it anyways, if that makes sense.

Reply:
Thanks for your rigorous test. Mostly this is for k8s platform  I think.  Will update.